sql/scan.go

```diff
@@ -424,6 +434,18 @@ func (n *scanNode) processKV(kv client.KeyValue) bool {
 			}
 		}
 	} else {
+		if n.pkVals != nil {
+			_, n.err = decodeKeyVals(n.pkVals, kv.ValueBytes())
+			if n.err != nil {
+				return false
+			}
+			for i, id := range n.desc.PrimaryIndex.ColumnIDs {
+				if qval := n.qvals[id]; qval != nil {
```

Comment:
nit: can we please be a little pedantic here and do the `ok` dance? i'd rather be alerted to the fact that a nil entry got into this map somehow than to save a few characters